src/control/controlcompressingproxy.cpp

```diff
@@ -0,0 +1,24 @@
+#include "control/controlcompressingproxy.h"
+
+#include "moc_controlcompressingproxy.cpp"
+
+// Compressor
+CompressingProxy::CompressingProxy(QObject* parent)
+        : QObject(parent){};
+
+bool CompressingProxy::emitCheck() {
+    m_lastEventRecursionOngoing = true;
```

Comment:
Why do you invoke this function recursively?

```
if (m_lastEventRecursionOngoing) {
    return true;
}
```

No need to call QCoreApplication::sendPostedEvents() repeatedly. Then the naming of m_lastEventRecursionOngoing also needs to be changed.

Reply:
I think we need to crawl through the event loop to receive the last value. With the proposed code we will receive the second oldest.

Reply:
Oh, right.